### ISSUE TITLE
Temp name collision 

### ISSUE THREAD

**Issue by biocyberman** [hCoV-2019/pangolin#43]
I started to jobs at the same time, and didn't choose a specific temp directory. This lead to name collision and 'race condition' in temp files: Run 1 creates temp files with /tmp/tax6358tax.aln.*
Run 2 at the same time can create the similar files. Specifying tempdir is a workaround, but it would be safer with unique temp file names.

**Comment by rambaut** [comment]
I suggest creating a directory in /tmp using the process id.

**Comment by aineniamh** [comment]
What I've done is just prepended a timestamp to the hash key, so that should be unique now between runs of pangolin.

**Comment by biocyberman** [comment]
Love how quickly you guys response :)

**Comment by tseemann** [comment]
@rambaut the correct way is to use the kernel  mktempd() system call, as PID is not unique when using shared /tmp on special cluster filesystems.

this can be done in python. I suggest a whole temp dir and put all output, temp calcs AND the .snakemake folder in it.

```
tempfile.TemporaryDirectory(suffix=None, prefix=None, dir=None)¶
This function securely creates a temporary directory using the same rules as mkdtemp(). The resulting object can be used as a context manager (see Examples). On completion of the context or destruction of the temporary directory object the newly created temporary directory and all its contents are removed from the filesystem.

The directory name can be retrieved from the name attribute of the returned object. When the returned object is used as a context manager, the name will be assigned to the target of the as clause in the with statement, if there is one.

The directory can be explicitly cleaned up by calling the cleanup() method.

Raises an auditing event tempfile.mkdtemp with argument fullpath.
```

**Comment by aineniamh** [comment]
This is now solved as described in #51
